Profiler/CProfilerCallback.cpp

```diff
@@ -1,488 +1,555 @@
+	EnterCriticalSection(&methodSetSynchronization);
+	writeFunctionInfosToLog();
+	LeaveCriticalSection(&methodSetSynchronization);
+	LeaveCriticalSection(&callbackSynchronization);
+	attachLog.logDetach();
+
+	traceLog.shutdown();
+	attachLog.shutdown();
+
+	if (this->ipc != NULL) {
+		delete ipc;
+		ipc = NULL;
+	}
+	if (config.shouldStartUploadDaemon()) {
+		createDaemon().notifyShutdown();
+	}
+	if (clrIsAvailable) {
+		profilerInfo->ForceGC();
+	}
+}
+
+HRESULT CProfilerCallback::Shutdown() {
+	try {
+		getShutdownGuard().shutdownInstance(true);
+	}
+	catch (...) {
+		handleException("Shutdown");
+	}
+	return S_OK;
+}
+
+void CProfilerCallback::adjustEventMask() {
+	DWORD dwEventMaskLow;
+	DWORD dwEventMaskHigh;
+	profilerInfo->GetEventMask2(&dwEventMaskLow, &dwEventMaskHigh);
+	dwEventMaskLow |= COR_PRF_MONITOR_ASSEMBLY_LOADS;
+
+	if (config.isTgaEnabled()) {
+		dwEventMaskLow |= COR_PRF_MONITOR_JIT_COMPILATION;
+		// disable force re-jitting for the light variant
+		if (!config.shouldUseLightMode()) {
+			dwEventMaskLow |= COR_PRF_DISABLE_ALL_NGEN_IMAGES;
+		}
+	}
+
+	if (config.isTiaEnabled()) {
+		dwEventMaskLow |= COR_PRF_MONITOR_ENTERLEAVE;
+		dwEventMaskLow |= COR_PRF_DISABLE_INLINING;
+	}
+
+	profilerInfo->SetEventMask2(dwEventMaskLow, dwEventMaskHigh);
+}
+
+HRESULT CProfilerCallback::AssemblyLoadFinished(AssemblyID assemblyId, HRESULT hrStatus) {
+	try {
+		return AssemblyLoadFinishedImplementation(assemblyId, hrStatus);
+	}
+	catch (...) {
+		handleException("AssemblyLoadFinished");
+		return S_OK;
+	}
+}
+
+HRESULT CProfilerCallback::AssemblyLoadFinishedImplementation(AssemblyID assemblyId, HRESULT hrStatus) {
+	if (!config.isProfilingEnabled()) {
+		return S_OK;
+	}
+	EnterCriticalSection(&callbackSynchronization);
+
+	int assemblyNumber = registerAssembly(assemblyId);
+
+	char assemblyInfo[BUFFER_SIZE];
+	int writtenChars = 0;
+
+	WCHAR assemblyName[BUFFER_SIZE];
+	WCHAR assemblyPath[BUFFER_SIZE];
+	ASSEMBLYMETADATA metadata;
+	getAssemblyInfo(assemblyId, assemblyName, assemblyPath, &metadata);
+
+	LeaveCriticalSection(&callbackSynchronization);
+
+	// Log assembly load.
+	writtenChars += sprintf_s(assemblyInfo + writtenChars, BUFFER_SIZE - writtenChars, "%S:%i",
+		assemblyName, assemblyNumber);
+
+	writtenChars += sprintf_s(assemblyInfo + writtenChars, BUFFER_SIZE - writtenChars, " Version:%i.%i.%i.%i",
+		metadata.usMajorVersion, metadata.usMinorVersion, metadata.usBuildNumber, metadata.usRevisionNumber);
+
+	if (config.shouldLogAssemblyFileVersion()) {
+		writtenChars += writeFileVersionInfo(assemblyPath, assemblyInfo + writtenChars, BUFFER_SIZE - writtenChars);
+	}
+
+	if (config.shouldLogAssemblyPaths()) {
+		sprintf_s(assemblyInfo + writtenChars, BUFFER_SIZE - writtenChars, " Path:%S", assemblyPath);
+	}
+	traceLog.logAssembly(assemblyInfo);
+
+	// Always return OK
+	return S_OK;
+}
+
+int CProfilerCallback::registerAssembly(AssemblyID assemblyId) {
+	int assemblyNumber = assemblyCounter;
+	assemblyCounter++;
```

Comment:
that's what `postfix++` does :)
```suggestion
	int assemblyNumber = assemblyCounter++;
```